### ISSUE TITLE
Upgrader:  Freezes, loses graphics, loses buttons, errors

### ISSUE THREAD

**Issue by sbulen** [SimpleMachines/SMF#3877]
Upgrader is behaving erratically today.  Brand new download from github.  Issues:
- it loses the 'continue' button
- it loses graphics
- it freezes
- error message about not finding english
- if I can get it to finish (sometimes enter works despite no continue button) I have received this message:
       * Notice: Undefined index: upgrade_upgrade_utility in (path)\upgrade.php on line 3334

![upgrader-lost-graphics-button](https://cloud.githubusercontent.com/assets/23568484/22359249/64e6bcb8-e3fa-11e6-93c7-c6d5f66edbfb.JPG)
![upgrader-lost-graphics-freezes](https://cloud.githubusercontent.com/assets/23568484/22359250/64eaf242-e3fa-11e6-8449-2ab7091d2cbd.JPG)
![upgrader-errors](https://cloud.githubusercontent.com/assets/23568484/22359248/64e648aa-e3fa-11e6-9fc2-419fb52672ef.JPG)

**Comment by Oldiesmann** [comment]
That undefined constant thing is my fault. Will fix that .

**Comment by Oldiesmann** [comment]
#3880 fixes the "undefined constant" error. That might fix some of the other issues as well, so try again and let me know what else is still broken.

**Comment by sbulen** [comment]
Still getting all these new errors.  I can't make it to the end to confirm the undefined constant error is OK.

**Comment by sbulen** [comment]
Looks like we got the Continue button back.  

The '30 seconds' message usually comes up when the DB fails.  I think we have a syntax error early in the script now.   
![image](https://cloud.githubusercontent.com/assets/23568484/22392032/2590c09e-e4aa-11e6-99e6-0fae410bdd76.png)

**Comment by sbulen** [comment]
Waitaminute...   I needed to restore my old Settings.php - didn't realize it got putzed...  Let me try again...

**Comment by sbulen** [comment]
Getting this error now:
You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '} } updateSettings(array('langList' => json_encode($newlangs)))' at line 1

![image](https://cloud.githubusercontent.com/assets/23568484/22392774/e6389072-e4b1-11e6-9d3f-34da43fca8a0.png)

**Comment by sbulen** [comment]
It's the apostrophe on line 2521

**Comment by sbulen** [comment]
It's the brace to indicate it's PHP not SQL...   Testing...

**Comment by sbulen** [comment]
Typo on 'memberes' too

**Comment by sbulen** [comment]
It hangs pretty consistently in this spot - may this task should be split up:
![image](https://cloud.githubusercontent.com/assets/23568484/22393565/94ae8d18-e4bd-11e6-9b82-ce5d3c7898f2.png)

**Comment by sbulen** [comment]
Still getting these issues, too:
![image](https://cloud.githubusercontent.com/assets/23568484/22393581/e31aeb9a-e4bd-11e6-9da5-732abb80f595.png)

**Comment by sbulen** [comment]
OK, got past:
  -  undefined constant
  -  a couple of db upgrade errors

Problems still outstanding:
  -  It still has lost graphics & the continue button
  -  This error:  Notice: Undefined index: upgrade_upgrade_utility in (path)\upgrade.php on line 3334
  -  The "server not responded" message near "add location column to calendar table"

If you're getting these errors, you can actually get it to complete by:
 - clicking on the 'click here' link allows the DB upgrade to get past the server not responded message
 - hitting enter works as a substitute for the continue button

**Comment by sbulen** [comment]
I do still get a 'server not responded' in the same location, but other than that, it just ran perfectly...   

Maybe a lot of the other issues were downstream effects of the bugs just addressed.  

I'm going to test further with a large site upgrade tomorrow.

**Comment by sbulen** [comment]
The issues depend on what you're upgrading:

On **_a fresh 2.0 install_**, the 2.1 upgrader works nearly perfectly.  The 'server not responded' still happens in that location, which is probably a real issue because it was basically empty.

On an upgrade against **_a mature, populated SMF 2.0 site_**, all of the other issues above still hold (button & header missing, all the text lookups missing).  You can force your way thru it & get a working site, but it's rough.

**Comment by sbulen** [comment]
The above PR fixes the freeze.

Hoping to have another PR shortly...   It's all tied to the -UTF8 suffix on the language files - there are a couple more checks to do.  The above issues will happen to a site that is already -utf8.  (The 2.0 vanilla install I used was not -utf8, I wanted to test the conversion by the upgrader.)

**Comment by sbulen** [comment]
I believe the 2 PRs above address all of the issues in this thread.

The PRs test out cleanly when upgrading new 2.0 installs as well as mature 2.0 installs.
